src/Common/src/Interop/OSX/Interop.CoreFoundation.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+using CFStringRef = System.IntPtr;
+using CFArrayRef = System.IntPtr;
+using FSEventStreamRef = System.IntPtr;
+using size_t = System.IntPtr;
+using FSEventStreamEventId = System.UInt64;
+using CFTimeInterval = System.Double;
+using CFRunLoopRef = System.IntPtr;
+
+internal static partial class Interop
+{
+    internal static partial class CoreFoundation
+    {
+        /// <summary>
+        /// This constant specifies that we want to use the default Run mode for the thread's Run loop.
+        /// </summary>
+        /// <remarks>
+        /// For more information, see the Apple documentation: https://developer.apple.com/library/mac/documentation/CoreFoundation/Reference/CFRunLoopRef/index.html
+        /// </remarks>
+        internal static IntPtr kCFRunLoopDefaultMode = CFStringCreateWithCString(IntPtr.Zero, "kCFRunLoopDefaultMode", 0);
+
+        /// <summary>
+        /// This is the default system path for the CoreFoundation library that PInvoke will use
+        /// </summary>
+        private const string CoreFoundationLibrary = "/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation";
+
+        /// <summary>
+        /// Tells the OS what encoding the passed in String is in. These come from the CFString.h header file in the CoreFoundation framework.
+        /// </summary>
+        internal enum CFStringBuiltInEncodings : uint
```

Comment:
In general, we want files under Interop to be fine-grained enough that if someone needs one thing from a file, they'll need pretty much everything from the file... we're trying to avoid the situation where someone imports a file and all of a sudden pulls in a ton of stuff they don't need.  Is that the case here? If yes, great.  If not, we should split this up.  More information is on the wiki at https://github.com/dotnet/corefx/wiki/Structuring-Interop-Code